contracts/lending/ttoken/strategies/compound/TTokenCompoundStrategy_2.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// contracts
+import {
+    RolesMods
+} from "../../../../contexts2/access-control/roles/RolesMods.sol";
+import { ADMIN } from "../../data.sol";
+
+// Interfaces
+import { ICErc20 } from "../../../../shared/interfaces/ICErc20.sol";
+import { TTokenStrategy } from "../TTokenStrategy.sol";
+import { ITToken_V3 } from "../../ITToken_V3.sol";
+
+// Libraries
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { NumbersLib } from "../../../../shared/libraries/NumbersLib.sol";
+import { LibMeta } from "../../../../shared/libraries/LibMeta.sol";
+import {
+    LibCompound
+} from "../../../../escrow/dapps/libraries/LibCompound.sol";
+
+// Storage
+import "../../token-storage.sol" as TokenStorage;
+import "./compound-storage.sol" as CompoundStorage;
+
+contract TTokenCompoundStrategy_2 is RolesMods, TTokenStrategy {
+    /**
+     * @dev it creates a reference to the TToken storage
+     */
+    function() pure returns (TokenStorage.Store storage)
+        private constant tokenStore = TokenStorage.store;
+
+    /**
+     * @dev it creates a reference to the Compound storage
+     */
+    function() pure returns (CompoundStorage.Store storage)
+        private constant compoundStore = CompoundStorage.store;
+
+    string public constant NAME = "CompoundStrategy_2";
+
+    uint32 constant BLOCKS_PER_YEAR = 2351330;
+
+    /* External Functions */
+
+    function totalUnderlyingSupply() external override returns (uint256) {
+        // Get current supply
+        uint256 currentSupply = tokenStore().underlying.balanceOf(
+            address(this)
+        ) + compoundStore().cToken.balanceOfUnderlying(address(this));
+        // Calculate bonus interest due
+        uint256 bonusInterest = ((currentSupply * 10) / 100) *
+            ((block.timestamp - compoundStore().lastBonusIntTimestamp) /
+                BLOCKS_PER_YEAR);
+        uint256 bonusSupply = tokenStore().underlying.balanceOf(
+            compoundStore().bonusGnosisSafe
+        );
+        // Deposit into
+        tokenStore().underlying.transferFrom(
+            compoundStore().bonusGnosisSafe,
+            address(this),
+            bonusInterest
+        );
+        ITToken_V3(address(this)).mint(bonusInterest);
```

Comment:
we actually need to remove this line. by not minting any additional TTokens and depositing additional underlying assets, we inherently are generating additional APY for existing lenders